File: client/src/app/models/conversation.ts
import { User } from "./activity";

export interface Conversation {
  id: string;
  creator: User;
  recipient: User;
  messageTo: string | null;
  messagesCount: number;
}

export interface ConversationItem {
  id: string;
  user: User;
  lastMessage: Message;
  messageTo: string;
  FriendId: string | null;
  MessagesCount: number;
}

export interface Message {
  id: string;
  createdAt: Date;
  body: string;
  author: User;
}

File: Domain/Conversation.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Conversation
    {
        public Guid Id { get; set; }
        public User Creator { get; set; }
        public User Recipient { get; set; }
        public ICollection<Message> Messages { get; set; } = new List<Message>();
        public int MessagesCount { get; set; } = 0;

#nullable enable
        public Guid? FriendId { get; set; }
        public Friend? Friend { get; set; }
        public Message? LastMessage { get; set; }

        public string? MessageTo { get; set; }

    }
}